File: server/src/main/java/com/fluffy/infra/storage/AwsS3ClientProperties.java
package com.fluffy.infra.storage;

import jakarta.validation.constraints.NotBlank;
import org.springframework.boot.context.properties.ConfigurationProperties;

@ConfigurationProperties(prefix = "spring.cloud.aws.s3")
public record AwsS3ClientProperties(
        @NotBlank String bucket,
        @NotBlank String domain
) {
}

File: server/src/main/kotlin/com/fluffy/infra/storage/AwsS3ClientProperties.kt
package com.fluffy.infra.storage

import jakarta.validation.constraints.NotBlank
import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "spring.cloud.aws.s3")
data class AwsS3ClientProperties(
    @field:NotBlank val bucket: String,
    @field:NotBlank val domain: String,
)